rust/src/pgsql/logger.rs

```diff
@@ -29,8 +29,13 @@ pub const PGSQL_LOG_PASSWORDS: u32 = BIT_U32!(0);
 fn log_pgsql(tx: &PgsqlTransaction, flags: u32, js: &mut JsonBuilder) -> Result<(), JsonError> {
     js.open_object("pgsql")?;
     js.set_uint("tx_id", tx.tx_id)?;
-    if let Some(request) = &tx.request {
-        js.set_object("request", &log_request(request, flags)?)?;
+    if !tx.requests.is_empty() {
+        js.open_object("request")?;
```

Comment:
Looks like we are missing calls to json builder array function here...

Reply:
it's more because so far the request object doesn't have to be an array, yet, as we shouldn't have duplicate messages -- even if we accept duplicate messages when parsing, we're consolidating all of them in a single message, for the request. But we're moving towards a v2 for the log, and then, yeah, this will require an array for the request log, too.